### PR TITLE
Code coverage requirements

### PR DESCRIPTION
**Pull request by Vegter**
Explain that in newer projects code coverage requirements are normally enforced

### CODE REVIEW

File: guides/back-end-projects.md [hunk 1]
```diff
@@ -58,8 +58,17 @@ For testing we prefer using [pytest](https://docs.pytest.org/), following its co
...
-Absolute code coverage requirements and enforcement thereof exist only in very rare cases.
+In earlier projects code coverage was normally not checked.
+Absolute code coverage requirements and enforcement thereof in these projects exist only in very rare cases.
+
+In newer projects code coverage is however normally checked. Absolute coverage requirements are part of these projects.
+
+### About code coverage
+
+Code coverage is a measure to define the quality and completeness of the test suite.
+The coverage should normally be at least 85% of statements and functions in each file.
```
Comment (bvhme) [inline guides/back-end-projects.md]: Is 85% an arbitrary number? If so I would recommend replacing it with something like `high` or something.

File: guides/back-end-projects.md [hunk 2]
```diff
@@ -58,8 +58,17 @@ For testing we prefer using [pytest](https://docs.pytest.org/), following its co
 
 However you may find the use of unittest in some (older) projects.
 
-Code coverage is normally not checked.
-Absolute code coverage requirements and enforcement thereof exist only in very rare cases.
+In earlier projects code coverage was normally not checked.
+Absolute code coverage requirements and enforcement thereof in these projects exist only in very rare cases.
+
+In newer projects code coverage is however normally checked. Absolute coverage requirements are part of these projects.
```
Comment (bvhme) [inline guides/back-end-projects.md]: Perhaps add `see the README of the project for more information` or something similar to indicate that different projects have different requirements and that those can be found there.

File: guides/back-end-projects.md [hunk 3]
```diff
@@ -58,8 +58,17 @@ For testing we prefer using [pytest](https://docs.pytest.org/), following its co
...
+
+### About code coverage
+
+Code coverage is a measure to define the quality and completeness of the test suite.
+The coverage should normally be at least 85% of statements and functions in each file.
+
+But beware: high coverage is not an absolute measure. Good tests may execute the same lines multiple times, both for the happy flow as for unexpected situations. Testing for just a high coverage is considered bad practice.
```
Comment (bvhme) [inline guides/back-end-projects.md]: In this perhaps we can talk about what is good practice, for instance something like 

> 'every bit of code code should be committed and PR'ed accompanied by the relevant tests. Good tests both test the code as well as describe what the code is meant to do and can often be referenced later in order to understand the intent of a commit'

**Review by bvhme — CHANGES_REQUESTED** [review]
Awesome, this makes me very happy

**Comment by bvhme** [comment]
( Also sorry this took so long @Vegter )

**Comment by Vegter** [comment]
@bvhme Improved documentation. Why we test and what we expect from it.

**Pull request merged**